### DISCUSSION TITLE
re-ID a repository

### DISCUSSION THREAD

**Discussion by lejeczek** [borgbackup/borg#7900] [General]
Hi guys.
I read here that a copy of a repo, when **borg** complains about the same ID, needs a new/different ID as a "fix"
I wonder - if this is "approved" way for such cases, @devel - why not add it, make it "official" in the toolset, as an arg/option.
Lastly - would there be a good/recommended method to generate new IDs with/for **borg**?
many thanks, L.

**Comment by ThomasWaldmann** [comment]
The repo id is in the repo config, so it can be changed easily.

But while that somehow works, it does of course not change the encryption key, so the crypto problem of nonce reuse (if you update the original repo and the copy of the repo independently) still remains.

So guess that mode of operations is nothing we want to encourage.

**Comment by callegar** [comment]
I am now in this condition, since I have found an old copy of some repo (that exists because of a disk upgrade) and I need to check if it contains something that was pruned from the current copy. Borg refuses to list the repo.

I have a few questions:

- could read only operations be allowed safely on repos that share the same id?
- is there a way to put a repo in a "read only mode" where the repo contents can be listed, restored, served on fuse, but the repo can not be updated in any way?

**Reply by ThomasWaldmann** [reply]
You want to avoid cache confusion. So just change the repo ID before using. You can also switch the repo to append-only.

Completely read-only is difficult, because borg always wants to create a lock in there.
